File: libtokens/src/operator_token.c
#include "operator_token.h"

// Operator Tokens
const char ADDITION  = '+';
const char ANNOTATION  = '@';
const char ASSIGNMENT  = '=';
const char COLON  = ':';
const char COMMA  = ',';
const char DIVISION  = '/';
const char DOLLAR_SIGN  = '$';
const char EXCLUSIVE_OR  = '^';
const char FIELD_ACCESS  = '.';
const char GRAVE_ACCENT  = '`';
const char GREATER_THAN  = '>';
const char LESS_THAN  = '<';
const char LOGICAL_CONJUCTION  = '&';
const char LOGICAL_DISJUNCTION  = '|';
const char LOGICAL_NEGATION  = '!';
const char MODULUS  = '%';
const char MULTIPLICATION  = '*';
const char NULL_COALESCING  = '?';
const char OCTOTHORPE  = '#';
const char SEMI_COLON  = ';';
const char SUBTRACTION  = '-';
const char TILDE  = '~';
const char UNDERSCORE  = '_';

// Operator Token Array
const char operators_array() {
    const char operators[23] = {
        ADDITION,
        ANNOTATION,
        ASSIGNMENT,
        COLON,
        COMMA,
        DIVISION,
        DOLLAR_SIGN,
        EXCLUSIVE_OR,
        FIELD_ACCESS,
        GRAVE_ACCENT,
        GREATER_THAN,
        LESS_THAN,
        LOGICAL_CONJUCTION,
        LOGICAL_DISJUNCTION,
        LOGICAL_NEGATION,
        MODULUS,
        MULTIPLICATION,
        NULL_COALESCING,
        OCTOTHORPE,
        SEMI_COLON,
        SUBTRACTION,
        TILDE,
        UNDERSCORE,
    };

    return *operators;
}

File: fortran-tokens/src/operator_token.f90
module operator_token
  implicit none

  !> Operator Tokens
  character(len=1), parameter :: ADDITION  = "+"
  character(len=1), parameter :: ANNOTATION  = "@"
  character(len=1), parameter :: ASSIGNMENT  = "="
  character(len=1), parameter :: COLON  = ":"
  character(len=1), parameter :: COMMA  = ","
  character(len=1), parameter :: DIVISION  = "/"
  character(len=1), parameter :: DOLLAR_SIGN  = "$"
  character(len=1), parameter :: EXCLUSIVE_OR  = "^"
  character(len=1), parameter :: FIELD_ACCESS  = "."
  character(len=1), parameter :: GRAVE_ACCENT  = "`"
  character(len=1), parameter :: GREATER_THAN  = ">"
  character(len=1), parameter :: LESS_THAN  = "<"
  character(len=1), parameter :: LOGICAL_CONJUCTION  = "&"
  character(len=1), parameter :: LOGICAL_DISJUNCTION  = "|"
  character(len=1), parameter :: LOGICAL_NEGATION  = "!"
  character(len=1), parameter :: MODULUS  = "%"
  character(len=1), parameter :: MULTIPLICATION  = "*"
  character(len=1), parameter :: NULL_COALESCING  = "?"
  character(len=1), parameter :: OCTOTHORPE  = "#"
  character(len=1), parameter :: SEMI_COLON  = ":"
  character(len=1), parameter :: SUBTRACTION  = "-"
  character(len=1), parameter :: TILDE  = "~"
  character(len=1), parameter :: UNDERSCORE  = "_"

end module operator_token

